bundles/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/bus/KNXBinding.java

```diff
@@ -108,7 +108,7 @@ public void removeKNXTypeMapper(KNXTypeMapper typeMapper) {
     @Override
     protected void internalReceiveCommand(String itemName, Command command) {
         logger.trace("Received command (item='{}', command='{}')", itemName, command.toString());
-        if (!isEcho(itemName, command)) {
+        if (!KNXConnection.getCompatibilityOH2() && !isEcho(itemName, command)) {
```

Comment:
@kaikreuzer, 
AHHRGG.. how simple, I should have seen. Thank You!!

See my tests KNX1.8 with *AutoUpdateBindingProvider* registered:

### OH1 TestSwitch.sendCommand(ON/OFF): *Received command* and a second *Received command* but no expected *Received update* per pass
```
2016-10-26 06:44:35.289 [TRACE] [.b.knx.internal.bus.KNXBinding] - Received command (item='TestSwitch', command='OFF')
2016-10-26 06:44:35.297 [TRACE] [.o.b.k.i.dpt.KNXCoreTypeMapper] - toTypeClass looking for dptId = 1.001
2016-10-26 06:44:35.300 [DEBUG] [.b.knx.internal.bus.KNXBinding] - Received groupWrite Event.
2016-10-26 06:44:35.300 [DEBUG] [.b.knx.internal.bus.KNXBinding] - Wrote value 'OFF' to datapoint 'command DP 1/1/10 TestSwitch, DPT main 0 id 1.001, low priority'
2016-10-26 06:44:35.301 [TRACE] [.o.b.k.i.dpt.KNXCoreTypeMapper] - toType datapoint DPT = 1.001
2016-10-26 06:44:35.308 [TRACE] [.b.knx.internal.bus.KNXBinding] - Added event (item='TestSwitch', type='OFF') to the ignore event list
2016-10-26 06:44:35.310 [TRACE] [.b.knx.internal.bus.KNXBinding] - Processed event (item='TestSwitch', type='OFF', destination='1/1/10')
2016-10-26 06:44:35.316 [TRACE] [.b.knx.internal.bus.KNXBinding] - Received command (item='TestSwitch', command='OFF')
2016-10-26 06:44:35.319 [TRACE] [.b.knx.internal.bus.KNXBinding] - We received this event (item='TestSwitch', state='OFF') from KNX, so we don't send it back again -> ignore!

2016-10-26 06:44:40.009 [TRACE] [.b.knx.internal.bus.KNXBinding] - Received command (item='TestSwitch', command='OFF')
2016-10-26 06:44:40.013 [TRACE] [.o.b.k.i.dpt.KNXCoreTypeMapper] - toTypeClass looking for dptId = 1.001
2016-10-26 06:44:40.015 [DEBUG] [.b.knx.internal.bus.KNXBinding] - Wrote value 'OFF' to datapoint 'command DP 1/1/10 TestSwitch, DPT main 0 id 1.001, low priority'
2016-10-26 06:44:40.015 [DEBUG] [.b.knx.internal.bus.KNXBinding] - Received groupWrite Event.
2016-10-26 06:44:40.017 [TRACE] [.o.b.k.i.dpt.KNXCoreTypeMapper] - toType datapoint DPT = 1.001
2016-10-26 06:44:40.020 [TRACE] [.b.knx.internal.bus.KNXBinding] - Added event (item='TestSwitch', type='OFF') to the ignore event list
2016-10-26 06:44:40.021 [TRACE] [.b.knx.internal.bus.KNXBinding] - Processed event (item='TestSwitch', type='OFF', destination='1/1/10')
2016-10-26 06:44:40.023 [TRACE] [.b.knx.internal.bus.KNXBinding] - Received command (item='TestSwitch', command='OFF')
2016-10-26 06:44:40.027 [TRACE] [.b.knx.internal.bus.KNXBinding] - We received this event (item='TestSwitch', state='OFF') from KNX, so we don't send it back again -> ignore!
```

### OH1 TestSwitch.postUpdate(ON/OFF): *Received update* and long AFTER that a unwanted *Received command* per pass
```
2016-10-26 06:45:55.290 [TRACE] [.b.knx.internal.bus.KNXBinding] - Received update (item='TestSwitch', state='OFF')
2016-10-26 06:45:55.297 [TRACE] [.o.b.k.i.dpt.KNXCoreTypeMapper] - toTypeClass looking for dptId = 1.001
2016-10-26 06:45:55.300 [DEBUG] [.b.knx.internal.bus.KNXBinding] - Received groupWrite Event.
2016-10-26 06:45:55.300 [DEBUG] [.b.knx.internal.bus.KNXBinding] - Wrote value 'OFF' to datapoint 'command DP 1/1/10 TestSwitch, DPT main 0 id 1.001, low priority'
2016-10-26 06:45:55.301 [TRACE] [.o.b.k.i.dpt.KNXCoreTypeMapper] - toType datapoint DPT = 1.001
2016-10-26 06:45:55.305 [TRACE] [.b.knx.internal.bus.KNXBinding] - Added event (item='TestSwitch', type='OFF') to the ignore event list
2016-10-26 06:45:55.308 [TRACE] [.b.knx.internal.bus.KNXBinding] - Processed event (item='TestSwitch', type='OFF', destination='1/1/10')
2016-10-26 06:45:55.309 [TRACE] [.b.knx.internal.bus.KNXBinding] - Received command (item='TestSwitch', command='OFF')
2016-10-26 06:45:55.315 [TRACE] [.b.knx.internal.bus.KNXBinding] - We received this event (item='TestSwitch', state='OFF') from KNX, so we don't send it back again -> ignore!

2016-10-26 06:46:00.019 [TRACE] [.b.knx.internal.bus.KNXBinding] - Received update (item='TestSwitch', state='ON')
2016-10-26 06:46:00.022 [TRACE] [.o.b.k.i.dpt.KNXCoreTypeMapper] - toTypeClass looking for dptId = 1.001
2016-10-26 06:46:00.024 [DEBUG] [.b.knx.internal.bus.KNXBinding] - Received groupWrite Event.
2016-10-26 06:46:00.024 [DEBUG] [.b.knx.internal.bus.KNXBinding] - Wrote value 'ON' to datapoint 'command DP 1/1/10 TestSwitch, DPT main 0 id 1.001, low priority'
2016-10-26 06:46:00.027 [TRACE] [.o.b.k.i.dpt.KNXCoreTypeMapper] - toType datapoint DPT = 1.001
2016-10-26 06:46:00.029 [TRACE] [.b.knx.internal.bus.KNXBinding] - Added event (item='TestSwitch', type='ON') to the ignore event list
2016-10-26 06:46:00.031 [TRACE] [.b.knx.internal.bus.KNXBinding] - Processed event (item='TestSwitch', type='ON', destination='1/1/10')
2016-10-26 06:46:00.032 [TRACE] [.b.knx.internal.bus.KNXBinding] - Received command (item='TestSwitch', command='ON')
2016-10-26 06:46:00.037 [TRACE] [.b.knx.internal.bus.KNXBinding] - We received this event (item='TestSwitch', state='ON') from KNX, so we don't send it back again -> ignore!
```

### OH2 TestSwitch.sendCommand(ON/OFF): *Received command* and as expected a *Received update* per pass
```
2016-10-26 07:03:35.004 [INFO ] [.internal.AutoUpdateProviderDelegate] - AutoUpdateProviderDelegate: autoUpdate run for 'TestSwitch'
2016-10-26 07:03:35.004 [INFO ] [.internal.AutoUpdateProviderDelegate] - AutoUpdateProviderDelegate: autoUpdate null
2016-10-26 07:03:35.005 [TRACE] [.binding.knx.internal.bus.KNXBinding] - Received command (item='TestSwitch', command='ON')
2016-10-26 07:03:35.005 [TRACE] [.binding.knx.internal.bus.KNXBinding] - Received update (item='TestSwitch', state='ON')
2016-10-26 07:03:35.006 [TRACE] [g.knx.internal.dpt.KNXCoreTypeMapper] - toTypeClass looking for dptId = 1.001
2016-10-26 07:03:35.007 [DEBUG] [.binding.knx.internal.bus.KNXBinding] - Wrote value 'ON' to datapoint 'command DP 1/1/10 TestSwitch, DPT main 0 id 1.001, low priority'
2016-10-26 07:03:35.008 [DEBUG] [.binding.knx.internal.bus.KNXBinding] - Received groupWrite Event.
2016-10-26 07:03:35.008 [WARN ] [.binding.knx.internal.bus.KNXBinding] - Ignoring local Event, received from my local Source address 15.15.244 for Group address 1/1/10.

2016-10-26 07:03:40.002 [INFO ] [.internal.AutoUpdateProviderDelegate] - AutoUpdateProviderDelegate: autoUpdate run for 'TestSwitch'
2016-10-26 07:03:40.002 [INFO ] [.internal.AutoUpdateProviderDelegate] - AutoUpdateProviderDelegate: autoUpdate null
2016-10-26 07:03:40.004 [TRACE] [.binding.knx.internal.bus.KNXBinding] - Received command (item='TestSwitch', command='OFF')
2016-10-26 07:03:40.004 [TRACE] [.binding.knx.internal.bus.KNXBinding] - Received update (item='TestSwitch', state='OFF')
2016-10-26 07:03:40.004 [TRACE] [g.knx.internal.dpt.KNXCoreTypeMapper] - toTypeClass looking for dptId = 1.001
2016-10-26 07:03:40.005 [DEBUG] [.binding.knx.internal.bus.KNXBinding] - Received groupWrite Event.
2016-10-26 07:03:40.005 [DEBUG] [.binding.knx.internal.bus.KNXBinding] - Wrote value 'OFF' to datapoint 'command DP 1/1/10 TestSwitch, DPT main 0 id 1.001, low priority'
2016-10-26 07:03:40.005 [WARN ] [.binding.knx.internal.bus.KNXBinding] - Ignoring local Event, received from my local Source address 15.15.244 for Group address 1/1/10.
```

### OH2 TestSwitch.postUpdate(ON/OFF): As expected a *Received update* per pass
```
2016-10-26 07:04:45.004 [TRACE] [.binding.knx.internal.bus.KNXBinding] - Received update (item='TestSwitch', state='ON')
2016-10-26 07:04:45.004 [TRACE] [g.knx.internal.dpt.KNXCoreTypeMapper] - toTypeClass looking for dptId = 1.001
2016-10-26 07:04:45.004 [DEBUG] [.binding.knx.internal.bus.KNXBinding] - Wrote value 'ON' to datapoint 'command DP 1/1/10 TestSwitch, DPT main 0 id 1.001, low priority'
2016-10-26 07:04:45.004 [DEBUG] [.binding.knx.internal.bus.KNXBinding] - Received groupWrite Event.
2016-10-26 07:04:45.004 [WARN ] [.binding.knx.internal.bus.KNXBinding] - Ignoring local Event, received from my local Source address 15.15.244 for Group address 1/1/10.

2016-10-26 07:04:50.003 [TRACE] [.binding.knx.internal.bus.KNXBinding] - Received update (item='TestSwitch', state='OFF')
2016-10-26 07:04:50.004 [TRACE] [g.knx.internal.dpt.KNXCoreTypeMapper] - toTypeClass looking for dptId = 1.001
2016-10-26 07:04:50.004 [DEBUG] [.binding.knx.internal.bus.KNXBinding] - Received groupWrite Event.
2016-10-26 07:04:50.004 [DEBUG] [.binding.knx.internal.bus.KNXBinding] - Wrote value 'OFF' to datapoint 'command DP 1/1/10 TestSwitch, DPT main 0 id 1.001, low priority'
2016-10-26 07:04:50.004 [WARN ] [.binding.knx.internal.bus.KNXBinding] - Ignoring local Event, received from my local Source address 15.15.244 for Group address 1/1/10.
```

## In OH1 KNX1.8 works really chaotic: 
To send a Command (rules sendCommand or Gui) first time is sent to KNX bus, then this Adress is blocked. No value update by sendCommand helps.
The only thing which can unblock is a postUpdate with a rule. postUpdate works as ecpected. After a postUpdate next sendCommand first time is sent to KNX bus, then this Adress is blocked again.

## In OH2 KNX1.8 works as expected.  Gui, sendCommand, postUpdate IN/OUT

## There is only an old known problem of the architecture of KNX1.8: 
The binding for writing KNX-Telegrams to calimero uses a function, which sets the KNX *READ-Flag*. 
The main side effect is, that values on the Bus are only sent after another or the same Groupadress was updated/sent on/from the KNX Bus.
Example: I want to send every second the value *OFF* to GA 1/1/10 to KNX Bus by a OH rule. First time when I send it to the KNX Bus, calimero will let pass it, second time it isn't, third time it isn't... . When any other GA is updated on KNX Bus (and comes through calimero) then ONLY NEXT TIME when I send my OFF to the KNX Bus, calimero will let pass it again, second time it isn't, third time it isn't... and so on.

But this is another issue, I just wanted to mention here again to avoid confusion for testers ;-)

What to do now?